packages/framework/src/Component/Elasticsearch/ElasticsearchStructureUpdateChecker.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\Elasticsearch;
+
+use Elasticsearch\Client;
+use Shopsys\FrameworkBundle\Component\ArrayUtils\RecursiveArraySorter;
+
+class ElasticsearchStructureUpdateChecker
+{
+    /**
+     * @var \Elasticsearch\Client
+     */
+    protected $client;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Elasticsearch\ElasticsearchStructureManager
+     */
+    protected $elasticsearchStructureManager;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\ArrayUtils\RecursiveArraySorter
+     */
+    protected $recursiveArraySorter;
+
+    /**
+     * @param \Elasticsearch\Client $client
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\ElasticsearchStructureManager $elasticsearchStructureManager
+     * @param \Shopsys\FrameworkBundle\Component\ArrayUtils\RecursiveArraySorter $recursiveArraySorter
+     */
+    public function __construct(
+        Client $client,
+        ElasticsearchStructureManager $elasticsearchStructureManager,
+        RecursiveArraySorter $recursiveArraySorter
+    ) {
+        $this->client = $client;
+        $this->elasticsearchStructureManager = $elasticsearchStructureManager;
+        $this->recursiveArraySorter = $recursiveArraySorter;
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $index
+     * @return bool
+     */
+    public function isNecessaryToUpdateStructure(int $domainId, string $index): bool
```

Comment:
I'm still a bit worried about the hard-coded keys in `prepareCurrentIndexDefinitionForComparing`. Can you perhaps create a functional test that recreates the structure (or migrates it, if that's better in your opinion) and then asserts that `isNecessaryToUpdateStructure` returns `false`?

Reply:
resolved in 19c21e422, 09503e5bb and e158d95a7